src/main/java/com/thedariusz/warnme/twitter/TweetService.java

```diff
@@ -22,10 +40,41 @@ public void syncTweets(String twitterUserId) {
         List<TweetDto> allTweets = twitterClient.fetchAllTweets(twitterUserId);
 
         List<MeteoAlert> meteoAlerts = allTweets.stream()
-                .filter(meteoAlertMapper::isMeteoAlert)
-                .map(meteoAlertMapper::mapToMeteoAlert)
+                .filter(this::isMeteoAlert)
+                .map(meteoAlertMapper::mapToMeteoAlertFromTweet)
                 .collect(Collectors.toList());
 
         meteoAlertService.save(meteoAlerts);
     }
+
+    boolean isMeteoAlert(TweetDto tweetDto) {
+        TweetType tweetType = getTweetTypeBasedOnHashTags(tweetDto.getHashTags());
+        return tweetType.equals(TweetType.METEO_ALERT);
+    }
+
+    TweetType getTweetTypeBasedOnHashTags(List<String> hashTags) {
```

Comment:
This method should be private as it doesn't do anything else outside of this class